docs/changes/units/17469.bugfix.rst

```diff
@@ -0,0 +1,3 @@
+Ensure that ``Unit.to`` also passes through array duck types that define
```

Comment:
is "array duck type" defined precisely anywhere ? I don't mind this expression being loosely used in discussions but I might be uncomfortable with including it in the changelog if there isn't a clear reference to point people to when they ask about it.

Reply:
I guess here it was mostly relevant that they are not numpy arrays. How about,
```
Ensure that ``Unit.to`` allows as ``value`` argument all array types that follow
the array API standard and define ``__array_namespace__``. Futhermore, for
backwards compatibility, generally pass through arguments that define a
``.dtype``, independent of whether that is a numpy data type.
```

Note that I really would like to get rid of `_condition_arg` altogether, or at least make it something that only triggers conversion on something positive, like `list`. But that would not be a bugfix...

Reply:
@neutrinoceros - would the above change to the changelog address your concern? If so, I'll push it up.

Reply:
works for me. Thanks !